### DISCUSSION TITLE
Directus (Docker) force create local database file

### DISCUSSION THREAD

**Discussion by Enhed** [directus/directus#12643] [Archive]
I created container with official image directus and add all nessesary envs for database. Directus creating all time volume with local database.sqlite and ingoring my db. How i can fix that?

**Comment by rijkvanzanten** [comment]
How are you adding the necessary environment variables? It sounds to me like it's defaulting to SQLite, because the db environment variables are missing somehow 🤔

**Reply by Enhed** [reply]
![image](https://user-images.githubusercontent.com/36981278/162702176-1d47f40a-4c3e-416d-9a83-d0397f336a0e.png)
variables is valid, locally directus can connect but container creating sqlite)

**Reply by azrikahar** [reply]
`DB_FILENAME` is set in the Dockerfile as seen here:

https://github.com/directus/directus/blob/985605b3564f8d9c0943dcd95ab7e08663c52435/docker/Dockerfile#L29-L33

in case `DB_CLIENT` isn't set at all. However since you have set `DB_CLIENT` as `pg`, the `DB_FILENAME` variable shouldn't be relevant here, even if it is seemingly set. 

Just to clarify, by `locally directus can connect`, you mean your original issue of "ingoring my db" is already resolved?

**Reply by Enhed** [reply]
> Just to clarify, by locally directus can connect, you mean your original issue of "ingoring my db" is already resolved?  

no, its just locally instance, i want to use docker official container in prod with my db)

**Reply by azrikahar** [reply]
It can be hard to understand what is the "prod" in question here as it can mean a number of various environments, but assuming you are using docker-compose judging from the use of `DB_HOST` being set as `postgresql` _only_, you may need to ensure that environment variable is actually valid.
